File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    
  const stack = [];

  for(let i=0;i<s.length;i++){
    const char = s[i];

    if(char === '(' || char === '{' || char === '[' ){
      stack.push(char)
    }
    else {
      if(char === ')'){
        if(stack.pop() !== '('){
          return false
        }
      } else if(char === '}'){
        if(stack.pop() !== '{'){
          return false
        }
      } else if(char === ']'){
        if(stack.pop() !== '['){
          return false
        }
      }

    }

  }

  return stack.length === 0

};

File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    
    def isValid(self, s: str) -> bool:
        
        stack = []
        par = {')':'(',']': '[', '}': '{'}
        
        for char in s:
            
            if char in ('(','[', '{'):
                
                stack.append(char)
            
            elif char in (')',']','}'):
                if not len(stack) :
                    return False
                if stack.pop() != par[char]:
                    return False
        
        if len(stack) > 0:
            return False
        return True
        
        
        